adapters_test.go

```diff
@@ -0,0 +1,240 @@
+		})
+	}
+
+	t.Run("returns error when input contains invalid binary data", func(t *testing.T) {
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{[]byte("invalid")})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unexpected number of messages in Cosmos tx", func(t *testing.T) {
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, &codectypes.Any{}, nil)
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when L1 Attributes tx is not found", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal MsgL1Txs message", func(t *testing.T) {
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, &codectypes.Any{Value: []byte("invalid")}, nil)
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when MsgL1Tx contains only non-deposit txs", func(t *testing.T) {
+		txBytes, _, _ := generateTxsAndTxsBytes(t, 0, 5)
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{TxBytes: txBytes})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal binary data within MsgL1Tx", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+			TxBytes: [][]byte{[]byte("invalid")},
+		})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+}
+
+func signTxHelper(t *testing.T, sdkTx *tx.Tx, signTx monomer.TxSigner) *tx.Tx {
+	if signTx == nil {
+		return sdkTx
+	}
+	err := signTx(sdkTx)
+	require.NoError(t, err)
+	return sdkTx
+}
+
+func generateDepositSDKMsgBytes(t *testing.T, msg *codectypes.Any, signTx monomer.TxSigner) []byte {
+	depositSDKMsgBytes, err := signTxHelper(t, &tx.Tx{Body: &tx.TxBody{Messages: []*codectypes.Any{msg}}}, signTx).Marshal()
+	require.NoError(t, err)
+	return depositSDKMsgBytes
+}
+
+func repeat[E any](x E, count int) []E {
+	newslice := make([]E, 0, count)
+	for range count {
+		newslice = append(newslice, x)
+	}
+	return newslice
+}
+
+func generateTxsAndTxsBytes(
+	t *testing.T,
+	depTxNum int,
+	nonDepTxNum int,
+) ([][]byte, ethtypes.Transactions, []hexutil.Bytes) {
+	_, depositTx, cosmosEthTx := testutils.GenerateEthTxs(t)
+	depositTxBytes, err := depositTx.MarshalBinary()
+	require.NoError(t, err)
+	cosmosEthTxBytes, err := cosmosEthTx.MarshalBinary()
+	require.NoError(t, err)
```

Comment:
You could just use `testutils.TxToBytes` here instead to avoid the err checking

```suggestion
	depositTxBytes := testutils.TxToBytes(depositTx)
	cosmosEthTxBytes := testutils.TxToBytes(cosmosEthTx)
```